src/field_type.cpp

```diff
@@ -124,18 +150,33 @@ void field_type::load( JsonObject &jo, const std::string & )
                   fallback_intensity_level.translucency );
         optional( jao, was_loaded, "convection_temperature_mod", intensity_level.convection_temperature_mod,
                   fallback_intensity_level.convection_temperature_mod );
-        optional( jao, was_loaded, "effect_id", intensity_level.field_effect.id,
-                  fallback_intensity_level.field_effect.id );
-        optional( jao, was_loaded, "effect_min_duration", intensity_level.field_effect.min_duration,
-                  fallback_intensity_level.field_effect.min_duration );
-        optional( jao, was_loaded, "effect_max_duration", intensity_level.field_effect.max_duration,
-                  fallback_intensity_level.field_effect.max_duration );
-        optional( jao, was_loaded, "effect_intensity", intensity_level.field_effect.intensity,
-                  fallback_intensity_level.field_effect.intensity );
-        optional( jao, was_loaded, "effect_body_part", intensity_level.field_effect.bp,
-                  fallback_intensity_level.field_effect.bp );
-        optional( jao, was_loaded, "inside_immune", intensity_level.field_effect.inside_immune,
-                  fallback_intensity_level.field_effect.inside_immune );
+        if( jao.has_array( "effects" ) ) {
+            JsonArray jae = jao.get_array( "effects" );
+            for( size_t j = 0; j < jae.size(); ++j ) {
+                JsonObject joe = jae.next_object();
+                field_effect fe;
+                mandatory( joe, was_loaded, "effect_id", fe.id );
+                optional( joe, was_loaded, "min_duration", fe.min_duration );
+                optional( joe, was_loaded, "max_duration", fe.max_duration );
+                optional( joe, was_loaded, "intensity", fe.intensity );
+                optional( joe, was_loaded, "body_part", fe.bp );
+                optional( joe, was_loaded, "is_environmental", fe.is_environmental );
+                optional( joe, was_loaded, "immune_in_vehicle", fe.immune_in_vehicle );
+                optional( joe, was_loaded, "immune_inside_vehicle", fe.immune_inside_vehicle );
+                optional( joe, was_loaded, "immune_outside_vehicle", fe.immune_outside_vehicle );
+                optional( joe, was_loaded, "chance_in_vehicle", fe.chance_in_vehicle );
+                optional( joe, was_loaded, "chance_inside_vehicle", fe.chance_inside_vehicle );
+                optional( joe, was_loaded, "chance_outside_vehicle", fe.chance_outside_vehicle );
+                fe.message = no_translation( jo.get_string( "message", "" ) );
+                fe.message_npc = no_translation( jo.get_string( "message_npc", "" ) );
```

Comment:
Calling `no_translation` calls `translation::translation( ..., no_translation_tag )`, which sets `translation::needs_translations` to `false` and stores the given string.
When you later call `translation::translated`, it checks `needs_translations`, sees it `false` and returns the raw (untranslated) string. And that raw string is taken directly from JSON.


> The translation is done on the fly using get_env_message/get_env_message_npc functions.

They call `translation::translated`, but you have constructed the `translation` object via `no_translation`, so it is considered an untranslatable string.

Reply:
Thanks. I see now I misunderstood the documentation.